hpx/lcos/local/spinlock_pool.hpp

```diff
@@ -14,39 +14,46 @@
 #define HPX_LCOS_LOCAL_SPINLOCK_POOL_HPP
 
 #include <hpx/config.hpp>
+#include <hpx/util/fibhash.hpp>
 #include <hpx/lcos/local/spinlock.hpp>
 
 #include <cstddef>
 
 ///////////////////////////////////////////////////////////////////////////////
 namespace hpx { namespace lcos { namespace local
 {
-#if HPX_HAVE_ITTNOTIFY != 0
     namespace detail
     {
-        template <typename Tag>
+#if HPX_HAVE_ITTNOTIFY != 0
+        template <typename Tag, std::size_t N>
         struct itt_spinlock_init
         {
             itt_spinlock_init();
             ~itt_spinlock_init();
         };
-    }
 #endif
+        // special struct to ensure cache line alignemnt
+        struct spinlock_holder
+        {
+            // pad to 64 bytes
+            char cacheline_pad[64 - sizeof(lcos::local::spinlock)];
```

Comment:
I would suggest we add an `hpx::topology::get_cache_line_size` and one for `page_size` as well. then there is a single place where all code can get the value from and it can be adjusted by platform/architecture in a less invasive manner than having 64 and 4096 in code scattered about.

Reply:
Good thinking!